h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -195,6 +199,6 @@ public int waitFor(long timeout) {
 
     @Override
     public void uncaughtException(Thread t, Throwable e) {
-        throw new UnsupportedOperationException("Uncaught exception is not supported by H2O tracker.");
+        stop();
```

Comment:
If `stop` throws an exception it will be ignored. We should decorate with try-catch and log or handle

Reply:
Is there any timeout for stop?